packages/dashboard/src/components/queryEditor/helpers/propertySelectionLabel.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  AssetPropertyExplorerProps,
+  AssetPropertyResource,
+} from '@iot-app-kit/react-components';
+import { isModeledPropertyInvalid } from './isModeledPropertyInvalid';
+import { DashboardWidget } from '~/types';
+
+export function propertySelectionLabel(
+  selectedItems: AssetPropertyExplorerProps['selectedAssetProperties'],
+  modeledDataStream: AssetPropertyResource,
+  selectedWidgets: DashboardWidget[]
+) {
+  const isPropertySelected = selectedItems?.find(
+    (item) => item.propertyId === modeledDataStream.propertyId
+  );
+
+  if (
+    isModeledPropertyInvalid(
+      modeledDataStream.dataType,
+      selectedWidgets?.at(0)?.type
+    )
+  ) {
+    return `${modeledDataStream.dataType} data not supported for the selected widget`;
```

Comment:
This first if statement does not depend on isPropertySelected. Rearrange the statements to put isPropertySelected after.